### ISSUE TITLE
Association is not defined (on removing)

### ISSUE THREAD

**Issue by dulgeoion** [strapi/strapi#1554]
**Informations**
- **Node.js version**: 10.5.0
- **npm version**: 5.6.0
- **Strapi version**: 12.7.1
- **Database**: pg
- **Operating system**: Mac

**What is the current behavior?**
So, what do we have. I had this issue before 12.7. When you try to delete some of your api with relations, in my case my model has relation 'one to many' and also it has file field. So, trying to delete some field, got this error .
![screen shot 2018-07-11 at 17 33 28](https://user-images.githubusercontent.com/15520510/42580147-939b2f46-8532-11e8-9472-37819c039021.png)

I don't know what exactly occurs this problem, relation 1xM or file field. I tried to delete this model without file associated to it, but still got that error. 

**Steps to reproduce the problem**
I have PG databas, model with associated file and relations: 1xM, 1x1. 

**What is the expected behavior?**
To make removing work.

**Comment by lauriejim** [comment]
Hello @dulgeoion .

1. Did you use migration guide or it's a fresh na alpha.12.7.1 app ?
1.1 If it's a migration, can you try with fresh new app to check if you can reproduce with last version

2. Can you try on your side if it's 1xM or file please (It will help me a lot, many issues to answer)

Thank you.

**Comment by dulgeoion** [comment]
Hi @lauriejim.
I had a talk with one more dev, who has the same problem as I have and we found that it occurs file uploading. So if you are not uploading file, you could do anything you want with model, but right after uploading associated file you face this issue while trying to update model or delete model. So problem is in files.

**Comment by merijnvanessen** [comment]
**request_id=: [2018-07-25T17:21:26.437Z] error association is not defined**

Hello. I have the same problem. I changed some content types on my development strapi.
Removed a few content types as well..

Now on the production strapi I can't update the  content types..
But when I added a new one.. I worked.. And i even can update the other types again

I'm on the nodejs v10.6.0
Strapi v3.0.0-alpha.12.6
